flax/linen/module.py

```diff
@@ -1655,8 +1655,18 @@ def loss(params, perturbations, inputs, targets):
       print(intm_grads['dense3']) # ==> [[-1.456924   -0.44332537  0.02422847]
                                   #      [-1.456924   -0.44332537  0.02422847]]
 
+    If perturbations are not passed to `apply`, `perturb` behaves like a no-op
+    so you can easily disable the behavior when not needed::
+
+      model.apply({'params': params, 'perturbations': perturbations}, x) # works as expected
+      model.apply({'params': params}, x) # behaves like a no-op
+
     """
-    value += self.variable(collection, name, lambda: jnp.zeros_like(value)).value # type: ignore
+    def _root_has_collection():
+      assert self.scope is not None
+      return collection in self.scope.root._variables
+    if self.is_mutable_collection(collection) or _root_has_collection():
```

Comment:
Maybe add a comment explaining this. I suppose the first clause handles initialization (since everything is mutable by default) and the second one apply (it will be ignored if you don't pass it).

Reply:
Done!